ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -1033,7 +1083,9 @@ for target in "${TARGET_RELEASES[@]}"; do
         cco_annotation
     fi
     if [[ "${UPGRADE_CCO_MANUAL_MODE}" == "oidc" ]]; then
-        update_cloud_credentials_oidc
+        if ! update_cloud_credentials_oidc; then
```

Comment:
Before this change, any non-zero return will lead the whole script exit as failure, but now seem like, the error was caught, and move forwards.

Reply:
BTW, there is some explicit exit in `update_cloud_credentials_oidc`, shall we set `UPGRADE_FAILURE_TYPE` before the exit inside the function ?

Reply:
Good catch. Because there are more than one exit in `update_cloud_credentials_oidc`, so I add the check outside of the function to avoid duplicated codes inside the func. I just updated to add `exit` code after setting UPGRADE_FAILURE_TYPE, the same with `rhel_post_upgrade`. PTAL again.

Reply:
Shall we also set `UPGRADE_FAILURE_TYPE` before `exit` inside `update_cloud_credentials_oidc` ?

Reply:
do u mean adding duplicated setting in each exit switch inside the function? it sounds like not necessary or do u have any other concern?

Reply:
Here is some local testing:
```
[jianlinliu@jialiu-laptop release]$ cat test.sh 
#!/bin/bash

set -o nounset
set -o errexit
set -o pipefail

function test1() {
    echo "testing"
    #ls /xxxx || return 1
    exit 1
    echo "done"
}

#test1
if ! test1; then
    echo "fail"
fi
[jianlinliu@jialiu-laptop release]$ sh -x test.sh 
+ set -o nounset
+ set -o errexit
+ set -o pipefail
+ test1
+ echo testing
testing
+ exit 1
```

So that means once the function execute `exit` line, there is no chance to set `UPGRADE_FAILURE_TYPE` in this `if` block. The `UPGRADE_FAILURE_TYPE` will use the default value - `overall`. Of course, if the default value is the expected type, that is not a problem.

Reply:
That is why generally I suggest to only use `return` inside function rather than `exit`

Reply:
Right, so the key point here is if we want to explicitly set  `UPGRADE_FAILURE_TYPE` for the exit lines inside `update_cloud_credentials_oidc` function. If we want those exit lines to use `overall` as `UPGRADE_FAILURE_TYPE`, ignore the comments.

Reply:
yeah, to identify the exit line in the code is the key point. currently we only add explicit failure type for those identified exit(such as `return 1`), for unclear exit, whether inside or outside function, it makes no difference. but there are code duplicates to add `export UPGRADE_FAILURE_TYPE=xxx` everywhere.  btw, for better understanding, we could have a huddle when you have time @jianlinliu wdyt?

Reply:
updated `exit` to `return` in `update_cloud_credentials_oidc`